src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -444,6 +474,17 @@ public void execute() throws MojoExecutionException {
 
             artifacts.add(project.getArtifact().getFile());
 
+            if (extraJars != null && !extraJars.isEmpty()) {
+                for (File extraJar : extraJars) {
+                    if (!Files.isRegularFile(extraJar.toPath())) {
+                        createErrorOutput();
```

Comment:
Please check if this error message are correct here 😄

Reply:
```java
private void createErrorOutput() {
        getLog().error("The project main artifact does not exist. This could have the following");
        getLog().error("reasons:");
....
```